### PR TITLE
Fix .bat launcher to allow special characters

### PR DESCRIPTION
**Pull request by sbc100**
Using `$*` inside the if/else blocks here was confusing the bat file
processor making it impossible to support commands that contains braces
and exclamation points.

There was an attempted fix made in #13723 which seemed hopefull but it
prevented the exclamation point.  Hopefully this will fix both cases.

Thanks to Mysoft on discord for the suggestion.

Fixes: #14063

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -10534,3 +10534,12 @@ def test_auto_ptr_cxx17(self):
       '-std=c++17',
       '-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR',
       '-Wno-deprecated-declarations'])
+
+  def test_special_chars_in_arguments(self):
+    # We has some regressions where the windows .bat files that run the compiler
+    # driver we failing to accept chars such as `(`, `)` and `!`
```
Comment (aheejin) [inline tests/test_other.py:10540]:
```suggestion
    # We have some regressions where the windows .bat files that run the compiler
    # driver failing to accept chars such as `(`, `)` and `!`
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -473,7 +473,7 @@ jobs:
       # note we do *not* build all libraries and freeze the cache; as we run
       # only limited tests here, it's more efficient to build on demand
       - run-tests:
-          test_targets: "other.test_emcc_cflags other.test_stdin other.test_bad_triple wasm2.test_sse1 wasm2.test_ccall other.test_closure_externs other.test_binaryen_debug other.test_js_optimizer_parse_error other.test_output_to_nowhere other.test_emcc_dev_null other.test_cmake* other.test_system_include_paths other.test_emar_response_file wasm2.test_utf16"
+          test_targets: "other.test_emcc_cflags other.test_stdin other.test_bad_triple wasm2.test_sse1 wasm2.test_ccall other.test_closure_externs other.test_binaryen_debug other.test_js_optimizer_parse_error other.test_output_to_nowhere other.test_emcc_dev_null other.test_cmake* other.test_system_include_paths other.test_emar_response_file wasm2.test_utf16 other.test_special_chars_in_arguments"
```
Comment (aheejin) [inline .circleci/config.yml:476]: What is this list of tests for?

**Review by aheejin — APPROVED** [review]
Nice! Thank you for fixing my failed attempt 😅

File: .circleci/config.yml (hunk 2 above)
Comment (sbc100) [reply .circleci/config.yml:476]: This is subset of tests that we run on windows during github CI.

**Comment by sbc100** [comment]
This failures are unrealted.. i think they are occurring because  `2.0.21 LTO RC3` just landed in emscripten releases and `./emsdk install tot` is treating that as the most recent build even though it is built from 2.0.21 which is several revisions ago.

**Pull request merged**